From b3c2712e266db74aaca9132363cbcd156a0a7b35 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:38 -0400
Subject: [infiniband] ocrdma: Fix traffic class shift

Message-id: <9720251a46767918000234f91168374476d36f81.1411571481.git.dledford@redhat.com>
Patchwork-id: 96607
O-Subject: [Patch RHEL7 03/48] RDMA/ocrdma: Fix traffic class shift
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

Use correct value for obtaining traffic class from device
response for Query QP request.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit a61d93d92f5c9533898098abb5f187840900aeb5)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index 4f8fb8b..b19effc 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -1414,7 +1414,7 @@ int ocrdma_query_qp(struct ib_qp *ibqp,
        OCRDMA_QP_PARAMS_HOP_LMT_MASK) >>
       OCRDMA_QP_PARAMS_HOP_LMT_SHIFT;
  qp_attr->ah_attr.grh.traffic_class = (params.tclass_sq_psn &
-           OCRDMA_QP_PARAMS_SQ_PSN_MASK) >>
+           OCRDMA_QP_PARAMS_TCLASS_MASK) >>
       OCRDMA_QP_PARAMS_TCLASS_SHIFT;
 
  qp_attr->ah_attr.ah_flags = IB_AH_GRH;
-- 
1.7.1